### ISSUE TITLE
Support new flag: `opam env --bare` for use with GitHub actions

### ISSUE THREAD

**Issue by shayne-fletcher** [ocaml/opam#5791]
Environment variables can be made available to steps in a GitHub workflow job by [writing the variable definitions to the `$GITHUB_ENV` file](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files).

- `opam env` prints an environment in terms of bash commands
- `opam env --sexp` prints an environment as an s-expression

If there existed a flag `--bare` (say) that printed an environment as exactly`VAR=VALUE` lines e.g.
```bash
  OPAM_SWITCH_PREFIX=/home/foo-user/.opam/default
  CAML_LD_LIBRARY_PATH=/home/foo-user/.opam/default/lib/ocaml/stublibs:/home/foo-user/.opam/default/lib/ocaml
  ...
```
that is, equivalent to
```bash
  opam env | sed -e "s/ export .*//g" -e "s/'//g" -e "s/\;//g"
```
then, a github action to configure an OPAM environment could be very simply written as just
```
  - run: |
    opam env --bare >> $GITHUB_ENV
```

**Comment by kit-ty-kate** [comment]
fair enough. We don’t have time to implement it until after the final release of opam 2.2.0 and even after larger works will most likely take priority but at least we’ll be more free to look at it.
If you want to have a stab at it (e.g. following the path of the `--sexp` option) you’re most than welcome to in the meantime
